### ISSUE TITLE
Adjust minimum version check for VS 2019 to avoid CMake 3.14

### ISSUE THREAD

**Issue by insinfo** [flutter/flutter#88589]
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):   CMake 3.15 or higher is required.  You are running version   3.14.19060802-MSVC_2

## Steps to Reproduce
```console
flutter create fsbackup
cd .\fsbackup\
flutter run -d Windows
```
**Expected results:** 
Run the program without error

**Actual results:** 
```console
flutter run -d Windows 
Downloading windows-x64/windows-x64-flutter tools...               27,3s
Downloading windows-x64/flutter-cpp-client-wrapper tools...        328ms
Downloading windows-x64-profile/windows-x64-flutter tools...        23,6s
Downloading windows-x64-release/windows-x64-flutter tools...        22,1s
Launching lib\main.dart on Windows in debug mode...
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.15 or higher is required.  You are running version
  3.14.19060802-MSVC_2


Building Windows application...
Exception: Unable to generate build files
```
###  flutter doctor -v    
```console
PS D:\MyDartProjects\fsbackup> flutter doctor -v     
[√] Flutter (Channel stable, 2.2.3, on Microsoft Windows [versÃ£o 10.0.19042.1165], locale pt-BR)
    • Flutter version 2.2.3 at C:\src\flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc1)
    • Android SDK at C:\Users\isaque\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)   
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.2.5)       
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.2.29306.81
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 48.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)   

[√] IntelliJ IDEA Community Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2019.3.4
    • Flutter plugin version 45.0.2
    • Dart plugin version 193.6911.31

[√] VS Code (version 1.55.2)
    • VS Code at C:\Users\isaque\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.22.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [versÃ£o 10.0.19042.1165]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 91.0.4472.77
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 92.0.902.73

• No issues found!
```

Visual Studio 2019 v16.0 MSVC 14.22.27905

**Comment by stuartmorgan** [comment]
I thought the lowest version of CMake shipped with VS 2019 was 3.15 when setting up the project, but it sounds like that's not the case. Does running the VS 2019 updater fix this?

If so, we should adjust the minimum version check (although I'm not finding a specific note of when they started shipping 3.15).

**Comment by insinfo** [comment]
I updated Visual Studio Community 2019 16.2.5 to Visual Studio Community 2019 16.11.1 and now it's working, it would be nice if the flutter doctor checks the minimal version

**Comment by cbracken** [comment]
Looking at some old [release notes](https://devblogs.microsoft.com/cppblog/visual-studio-cmake-support-clang-llvm-cmake-3-14-vcpkg-and-performance-improvements/) looks like Visual Studio 2019 Update 1 shipped with CMake 3.14.

I'll take a quick look at the [CMake 3.15 Release Notes](https://cmake.org/cmake/help/latest/release/3.15.html) and assuming we're not relying on anything that requires >=3.15, will send a patch dropping the requirement in the tool.

**Comment by cbracken** [comment]
This was landed.
